yt/utilities/lib/pixelization_routines.pyx

```diff
@@ -1233,6 +1298,51 @@ def pixelize_sph_kernel_arbitrary_grid(np.float64_t[:, :, :] buff,
 
                         buff[xi, yi, zi] += coeff * kernel_func(q)
 
+@cython.initializedcheck(False)
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+def pixelize_sph_kernel_gather_arbitrary_grid(np.float64_t[:, :, :] buff,
+        np.int64_t[:, :, :, :] pids, np.float64_t[:, :, :, :] dists,
+        np.float64_t[:] hsml, np.float64_t[:] pmass, np.float64_t[:] pdens,
+        np.float64_t[:] quantity_to_smooth, PyKDTree tree=None, np.int64_t
+        offset=0, kernel_name="cubic"):
+
+    cdef np.intp_t xsize, ysize, zsize
+    cdef np.float64_t w_j, coeff
+    cdef np.int64_t xi, yi, zi, pi
+    cdef np.float64_t q, h_j2, ih_j2
+    cdef int count, i, j, k, particle
+    cdef np.int64_t[:] tree_id
+
+    if tree is not None:
+        tree_id = tree.idx.astype("int64")
+
+    xsize, ysize, zsize = buff.shape[0], buff.shape[1], buff.shape[2]
+
+    kernel_func = get_kernel_func(kernel_name)
+
+    with nogil:
+        for xi in range(xsize):
+            for yi in range(ysize):
+                for zi in range(zsize):
+                    h_j2 = dists[xi, yi, zi, 0]
+                    ih_j2 = 1/h_j2
+
+                    for pi in range(pids.shape[3]):
+                        if tree is not None:
+                            particle = tree_id[pids[xi, yi, zi, pi]] - offset
+                            if(particle < 0 or particle > pmass.shape[0]):
+                                continue
+                        else:
+                            particle = pids[xi, yi, zi, pi]
+
+                        w_j = pmass[particle] / pdens[particle] / hsml[particle]**3
+                        coeff = w_j * quantity_to_smooth[particle]
```

Comment:
Sorry to be pedantic again -- but for readability we would typically reserve _w_ for the kernel (i.e. the result of kernel_func * 1/h^3). I know the terminology in Price's paper mixes these, but I would always stick with eqn. 25-style notation.

I would replace this with
```
prefactor_ij = pmass[particle] / pdens[particle] / hsml[particle]**3
q_ij = math.sqrt(dists[xi, yi, zi, pi]*ih_j2)
smoothed_quantity_j = prefactor_ij * quantity_to_smooth[particle] * kernel_func(q_ij)

buff[xi, yi, zi] += smoothed_quantity_j 
```
Are these _all_ just particle j's (i.e. the neighbour) quantities?

Reply:
Yes, they are all quantities from the same neighbor.

That is a good argument, there is deviation from typical SPH notation. I think if I make this change, then it would be good to make that change to every pixelization function in this file to preserve consistency (currently, the notation in this function is exactly the same as every other function in this file). Thoughts on that @ngoldbaum?